Sources/Turf/Geometries/LineString.swift

```diff
@@ -308,4 +308,25 @@ extension LineString {
         // Ported from https://github.com/Turfjs/turf/blob/4e8342acb1dbd099f5e91c8ee27f05fb2647ee1b/packages/turf-simplify/lib/simplify.js
         coordinates = Simplifier.simplify(coordinates, tolerance: tolerance, highestQuality: highestQuality)
     }
+    
+    /**
+     Returns all intersections with another `LineString`.
+     
+     This function is roughly equivalent to the [turf-line-intersect](https://turfjs.org/docs/#lineIntersect) package of Turf.js ([source code](https://github.com/Turfjs/turf/tree/master/packages/turf-line-intersect/)).
+     */
+    public func intersection(with line: LineString) -> [LocationCoordinate2D]? {
+        var intersections: [String : LocationCoordinate2D] = [:]
+        zip(coordinates.dropLast(), coordinates.dropFirst()).forEach { segment1 in
+            zip(line.coordinates.dropLast(), line.coordinates.dropFirst()).forEach { segment2 in
+                if let intersection = Turf.intersection(LineSegment(segment1.0, segment1.1),
+                                                        LineSegment(segment2.0, segment2.1)) {
+                    let key = "\(intersection.latitude),\(intersection.longitude)"
+                    if intersections[key] == nil {
+                        intersections[key] = intersection
+                    }
+                }
+            }
+        }
+        return intersections.isEmpty ? ([LocationCoordinate2D]?).none : Array(intersections.values)
```

Comment:
If there are no intersections, return an empty array instead of `nil`, to simplify the set of possible return values.